Sources/Sentry/SentrySystemEventsBreadcrumbs.m

```diff
@@ -0,0 +1,174 @@
+#import "SentrySystemEventsBreadcrumbs.h"
+#import "SentrySDK.h"
+#import "SentryLog.h"
+
+// all those notifications are not available for tvOS
+#if TARGET_OS_IOS
+#    import <UIKit/UIKit.h>
+#endif
+
+@implementation SentrySystemEventsBreadcrumbs
+
+- (void)start
+{
+#if TARGET_OS_IOS
+    UIDevice *currentDevice = [UIDevice currentDevice];
+    if (currentDevice == nil) {
+        [SentryLog logWithMessage:@"currentDevice is null, it won't be able to record battery breadcrumbs."
+                         andLevel:kSentryLogLevelDebug];
+        return;
+    }
+    
+    [self initBatteryObserver:currentDevice];
+    [self initOrientationObserver:currentDevice];
+    [self initKeyboardVisibilityObserver];
+    [self initScreenshotObserver];
+    
+#else
+    [SentryLog logWithMessage:@"NO UIKit -> [SentrySystemEventsBreadcrumbs.start] does nothing."
+                     andLevel:kSentryLogLevelDebug];
+#endif
+}
+
+#if TARGET_OS_IOS
+- (void)initBatteryObserver:(UIDevice*)currentDevice
+{
+    if (currentDevice.batteryMonitoringEnabled == NO) {
+        currentDevice.batteryMonitoringEnabled = YES;
+    }
+    
+    NSNotificationCenter *defaultCenter = [NSNotificationCenter defaultCenter];
+    
+    // https://developer.apple.com/documentation/uikit/uidevicebatteryleveldidchangenotification
+    [defaultCenter addObserver:self selector:@selector(batteryStateChanged:) name:UIDeviceBatteryLevelDidChangeNotification object:currentDevice];
+    // https://developer.apple.com/documentation/uikit/uidevicebatterystatedidchangenotification
+    [defaultCenter addObserver:self selector:@selector(batteryStateChanged:) name:UIDeviceBatteryStateDidChangeNotification object:currentDevice];
```

Comment:
`h`: Why do we have this twice here? If there is a reason, we should add a comment.

Reply:
they are not the same
`name:UIDeviceBatteryLevelDidChangeNotification` != `name:UIDeviceBatteryStateDidChangeNotification`

Reply:
Ops, they looked the same to me. Sorry